tests/ops/models/test_datasetconfig.py

```diff
@@ -200,41 +205,214 @@ def test_validate_dataset_reference_invalid(db: Session, dataset_config: Dataset
 
     dataset_key = dataset_config.fides_key
-    collection_name = dataset_config.dataset["collections"][0]["name"]
+    collection_name = dataset_config.ctl_dataset.collections[0]["name"]
     field_name = "fake_field"
     dsr = FidesDatasetReference(dataset=dataset_key, field=f"{collection_name}")
     with pytest.raises(ValidationError) as e:
         validate_dataset_reference(db, dsr)
     assert "must include at least two dot-separated components" in e.value.message
 
     dataset_key = dataset_config.fides_key
-    collection_name = dataset_config.dataset["collections"][0]["name"]
+    collection_name = dataset_config.ctl_dataset.collections[0]["name"]
     field_name = "fake_field"
     dsr = FidesDatasetReference(dataset=dataset_key, field=f".")
     with pytest.raises(ValidationError) as e:
         validate_dataset_reference(db, dsr)
     assert "must include at least two dot-separated components" in e.value.message
 
     dataset_key = dataset_config.fides_key
-    collection_name = dataset_config.dataset["collections"][0]["name"]
+    collection_name = dataset_config.ctl_dataset.collections[0]["name"]
     field_name = "fake_field"
     dsr = FidesDatasetReference(dataset=dataset_key, field="")
     with pytest.raises(ValidationError) as e:
         validate_dataset_reference(db, dsr)
     assert "must include at least two dot-separated components" in e.value.message
+
+
+class TestUpsertWithCtlDataset:
+    def test_no_existing_dataset_config_or_ctl_dataset(
+        self, db, example_datasets, connection_config
+    ):
+        """Ctl Dataset is created"""
+        postgres_dataset = example_datasets[0]
+
+        dataset_config = DatasetConfig.upsert_with_ctl_dataset(
+            db=db,
+            data={
+                "connection_config_id": connection_config.id,
+                "fides_key": "brand_new_fides_key",
+                "dataset": postgres_dataset,
+            },
+        )
+        assert dataset_config.fides_key == "brand_new_fides_key"
+        assert dataset_config.ctl_dataset_id is not None
+
+        ctl_dataset = dataset_config.ctl_dataset
+
+        assert (
+            ctl_dataset.fides_key == "postgres_example_test_dataset"
+        ), "New ctl dataset record created"
+        assert ctl_dataset.description == postgres_dataset["description"]
+        assert ctl_dataset.organization_fides_key == "default_organization"
+        assert ctl_dataset.data_categories == postgres_dataset.get("data_categories")
+        assert ctl_dataset.collections is not None
+
+        dataset = dataset_config.dataset
+        assert dataset["description"] == postgres_dataset["description"]
+        assert (
+            dataset.get("organization_fides_key") is None
+        ), "Existing behavior, DatasetConfig.dataset not validated by Fideslang Dataset first"
+        assert dataset["collections"] is not None
+
+        dataset_config.delete(db)
+        ctl_dataset.delete(db)
```

Comment:
Are these required? IIRC we have a fixture that runs after every test that clears out all of the tables.

https://github.com/ethyca/fides/blob/e77d6dc6fc7e56b269b77bf94491fa41ad7678a7/tests/ops/conftest.py#L107-L132

Reply:
Not technically, but I think this fixture that clears all the tables is too aggressive because there are resources that are expected to be in the database. 

 I filed an issue to investigate this further and in the meantime trying to make my new tests more self-sufficient generally https://github.com/ethyca/fides/issues/2016